### ISSUE TITLE
The `changes` command should wait `ready_for_reads`

### ISSUE THREAD

**Issue by coffeemug** [rethinkdb/rethinkdb#4893]
This came out of the discussion in https://github.com/rethinkdb/rethinkdb/issues/4754#issuecomment-143445619.

I think the `changes` command should internally wait for `ready_for_reads` when it's first called (as opposed to throwing).

We don't typically wait on the cluster in other queries, because if the cluster isn't ready and the query fails, the end user can just "rerun" it by refreshing the page, clicking a button again, etc. This way the developer can pretty much ignore the issue in the code (other than handling the exception and propagating the error message to the end user).

However, this isn't the case with changefeeds. If a node fails, the developer has to recreate the feed, and therefore any time anybody uses the `changes()` command, the proper way to write the code is to prepend it with a `wait`. This sounds unnecessarily painful and complicated -- the user shouldn't have to do that. Which is why I think `changes()` should block at the beginning to make sure the cluster is ready. Opinions?

/cc @mlucy @danielmewes

**Comment by danielmewes** [comment]
I might be thinking about this wrong, but I'm not sure if this will help to simplify any code for the user.
The changefeed might still fail after it has been started and has to be recreated. So all the code for retrying - with the exception of the `wait` call - still has to exist, doesn't it?

I'm also not sure how big the difference between changefeeds and regular queries really is in this respect. It probably depends a lot on how changefeeds are used in the application.

The implicit `wait` would avoid unintentionally getting into busy waiting when retrying the changefeed creation over and over.
Though it also involves the risk of hiding an issue and making it seem like the changefeed was created successfully, while it's really waiting for the table to become available. The user wouldn't see any changes during that time.

**Comment by coffeemug** [comment]
> So all the code for retrying - with the exception of the `wait` call - still has to exist, doesn't it?

Yes. That still seems easier, though. If we want feeds to be dead simple, it sounds like something we should solve. (Eventually we should also solve the retrying part, but I think even small steps make a big difference)

I agree with the problems you pointed out above; I'm not even sure that implicitly waiting is the right thing to do. But writing robust code right now *is* too complex for users (you have to loop and wait for every feed), and I think we need to find a way to simplify this process.
